backend/inspirehep/records/tasks.py

```diff
@@ -43,3 +56,49 @@ def update_records_relations(uuids):
 
     db.session.commit()
     return uuids
+
+
+@shared_task(ignore_result=False, bind=True, queue="migrator")
+def recalculate_references_for_merged_record(self, merged_record_uri, record_uri):
+    merged_record_schema = merged_record_uri.rsplit("/", 2)[1]
+    if merged_record_schema == "literature":
+        merged_record_schema = "hep"
+    referenced_records_indices_and_paths = get_refs_to_schemas()[merged_record_schema]
+    match_reference_with_config(
+        referenced_records_indices_and_paths, merged_record_uri, record_uri
+    )
+
+
+def match_reference_with_config(config_list, merged_record_uri, new_record_uri):
```

Comment:
this is a misnomer, as it does much more than matching. I think you need to step back and think about what you need to do from a high-level perspective, create functions for the different steps, then write an implementation for those functions (recursively). We can discuss this further if needed.